File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length <= 1) return nums.length;

        HashSet<Integer> numSet = new HashSet<>();
        for (int num : nums) {
            numSet.add(num);
        }

        int ans = 1;

        for (int num : nums) {
            if (numSet.contains(num - 1)) {
                continue;
            }

            int curr = num;
            int seq = 0;
            while (numSet.contains(curr)) {
                seq++;
                curr++;
                ans = Math.max(ans, seq);
            }

            if (ans > nums.length / 2) {
                break;
            }
        }
        return ans;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return len(nums)

        num_set = set(nums)
        longest = 1

        for num in num_set:
            # can you find num-1? if so this is not the start of the sequence, skip
            if num - 1 in num_set:
                continue

            curr = num
            seq = 0
            while curr in num_set:
                curr += 1
                seq += 1
                longest = max(seq, longest)

        return longest